lib/rucio/core/replica_sorter.py

```diff
@@ -179,9 +180,19 @@ def sort_replicas(dictreplica: "Dict", client_location: "Dict", selection: "Opti
     :param default: the default sorting algorithm (random, if not defined).
     :returns: the keys of dictreplica in a sorted list.
     """
+    if len(dictreplica) == 0:
+        return []
     if not selection:
         selection = 'geoip'
 
+    items = [(key, value) for key, value in dictreplica.items()]
```

Comment:
Is this any different to `list(dictreplica.items())`?

Reply:
good question, have to test that :)

Reply:
while the standard documentation mentions that `dict.items()` *should* return an iterable of tuples, I'd rather be on the safe side here and make the tuples explicit.